plugin/rewrite/ttl.go

```diff
@@ -9,81 +9,90 @@ import (
 
 	"github.com/coredns/coredns/plugin"
 	"github.com/coredns/coredns/request"
-	//"github.com/miekg/dns"
+
+	"github.com/miekg/dns"
 )
 
+type ttlResponseRule struct {
+	TTL uint32
+}
+
+func (r *ttlResponseRule) RewriteResponse(rr dns.RR) {
+	rr.Header().Ttl = r.TTL
+}
+
 type exactTTLRule struct {
 	NextAction string
 	From       string
-	ResponseRule
+	response   *ttlResponseRule
```

Comment:
why do these need to be pointers now? I think added a NoopRule can also remove the need to make `nil` the thing that conveys 'noop'. Would also make this PR much smaller, 'cause these changes can all be reverted

Reply:
I'll use the same pattern with a `ttlRuleBase` as for the name rules, which simplifies the complete rule implementations.